src/main/java/mate/academy/service/impl/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,48 @@
+package mate.academy.service.impl;
+
+import java.util.List;
+import mate.academy.dao.ShoppingCartDao;
+import mate.academy.dao.TicketDao;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.MovieSession;
+import mate.academy.model.ShoppingCart;
+import mate.academy.model.Ticket;
+import mate.academy.model.User;
+import mate.academy.service.ShoppingCartService;
+
+@Service
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    @Inject
+    private ShoppingCartDao shoppingCartDao;
+    @Inject
+    private TicketDao ticketDao;
+
+    @Override
+    public void addSession(MovieSession movieSession, User user) {
+        Ticket ticket = new Ticket(movieSession, user);
+        ticketDao.add(ticket);
+
+        ShoppingCart currentCart = getByUser(user);
+        currentCart.getTickets().add(ticket);
+        shoppingCartDao.update(currentCart);
+    }
+
+    @Override
+    public ShoppingCart getByUser(User user) {
+        return shoppingCartDao.getByUser(user).get();
+    }
+
+    @Override
+    public void registerNewShoppingCart(User user) {
+        shoppingCartDao.add(new ShoppingCart(user, List.of()));
+
+    }
+
+    @Override
+    public void clear(ShoppingCart shoppingCart) {
+        shoppingCart.setTickets(List.of());
```

Comment:
```suggestion
        shoppingCart.setTickets(new ArrayList<>());
```

Lets use 'new ArrayList<>()' here as List.of() returns an unmodifiable list